ci/test_cpp.sh

```diff
@@ -14,7 +14,9 @@ rapids-logger "Generate C++ testing dependencies"
 rapids-dependency-file-generator \
   --output conda \
   --file-key test_cpp \
-  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch)" | tee env.yaml
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch)" \
+  --channel "${CPP_CHANNEL}" \
+  | tee env.yaml
```

Comment:
Looks like this one wasn't finished. The download of packages needs to be moved up to before `rapids-mamba-retry env create`, and the `rapids-mamba-retry install` below should be removed.